File: Lab 6/C++/main.cpp
#include "Container.h"
#include <iostream>

int main() {
    Container myText;// Створюємо об'єкт контейнера, який буде зберігати рядки

    myText.addLine("Hallo");
    myText.addLine("My name Artem, 17 y.o.");
    myText.addLine("I need 3120301 dollars");
    myText.addLine("This line should be removed at the end");

    std::cout << "--------------------All text--------------------\n";
    myText.showAll();

    std::cout << "\n--------------------Result--------------------" << std::endl;

    //std::cout << "\nLonger string: " << myText.getLongestLine() << std::endl; помилка     повертає вказівник на рядок а не сам рядок
    const Text* longest = myText.getLongestLine();
    if (longest) {
        std::cout << "\nLonger string: " << longest->getLine() << std::endl;
    }
    
    std::cout << "Total number of symbols: " << myText.getTotalLength() << std::endl;
    std::cout << "Percent numbers in text: " << myText.getNumberPercent() << "%\n";

    std::cout << "\n----------------After remove line----------------" << std::endl;
    myText.removeLine(3);
    myText.showAll();

    return 0;
}

File: Lab 6/C#/main.cs
using System;

class Program
{
    static void Main()
    {
        Container myText = new Container(); // Створюємо об'єкт контейнера

        myText.AddLine("Hallo");
        myText.AddLine("My name Vasya, 17 y.o.");
        myText.AddLine("I need 3120301 dollars");
        myText.AddLine("This line should be removed at the end");

        Console.WriteLine("--------------------All text--------------------");
        myText.ShowAll();

        Console.WriteLine("\n--------------------Result--------------------");

        Text longest = myText.GetLongestLine();
        if (longest != null)
        {
            Console.WriteLine("\nLonger string: " + longest.GetLine());
        }

        Console.WriteLine("Total number of symbols: " + myText.GetTotalLength());
        Console.WriteLine("Percent numbers in text: " + myText.GetNumberPercent() + "%");

        Console.WriteLine("\n----------------After remove line----------------");
        myText.RemoveLine(3); // Видаляємо останній рядок
        myText.ShowAll();
    }
}

